File: dailybyte/03_vaccum_cleaner_route/vaccum.rb
def vaccum(str)
    x = 0
    y = 0
    str.each_char do |char|
        if char == "L"
            x -= 1
        elsif char == "R"
            x += 1
        elsif char == "U"
            y += 1
        elsif char == "D"
            y -= 1
        end
    end
    return x == 0 && y == 0
end

puts vaccum("LR") # true
puts vaccum("URURD") # false
puts vaccum("RUULLDRD") # true

File: dailybyte/03_vaccum_cleaner_route/vaccum.py
def vaccum_cleaner_route(route):
    x = 0
    y = 0
    for direction in route:
        if direction == "L":
            x -= 1
        elif direction == "R":
            x += 1
        elif direction == "U":
            y += 1
        elif direction == "D":
            y -= 1
    return x == 0 and y == 0


print(vaccum_cleaner_route("LR"))
print(vaccum_cleaner_route("URURD"))
print(vaccum_cleaner_route("RUULLDRD"))